### PR TITLE
[KotlinCleanup] AnkiDroidJsAPI.kt

### PR DESCRIPTION
**Pull request by Jatin887**
## Purpose / Description
Kotlin CleanUp

## Fixes
Part of [https://github.com/ankidroid/Anki-Android/issues/10489]

## Approach
Using lint and some code changes which were mentioned in the comments and required to make the code more reliable

## How Has This Been Tested?
Physical Device One Plus Nord2

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPI.kt [hunk 1]
```diff
@@ -42,11 +42,11 @@ import com.ichi2.libanki.Decks
 import com.ichi2.libanki.SortOrder
 import com.ichi2.utils.JSONException
 import com.ichi2.utils.JSONObject
-import com.ichi2.utils.KotlinCleanup
 import timber.log.Timber
 
-@KotlinCleanup("replace activity.currentCard!! with property")
 open class AnkiDroidJsAPI(private val activity: AbstractFlashcardViewer) {
+    private val currentCard = activity.currentCard!!
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPI.kt:48]: this needs to be a getter, not a variable

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPI.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPI.kt]: Yes

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidJsAPI.kt]: Added

**Review by david-allison — APPROVED** [review]
Looks great!

**Review by mikehardy — APPROVED** [review]
nice!

**Comment by mikehardy** [comment]
Hi there! APRIL OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [April 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-04-01..2022-04-30)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for all contributions. The only PRs that will not go through the OpenCollective process are ones directly related to am accepted GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**